src/nash.jl

```diff
@@ -46,10 +46,6 @@ end
 function solve_nash(hr1::HMatrix, hr2::HMatrix)
     NEs = NTuple{2,Vector{Rational{BigInt}}}[]
 
-    # Step 1
-    FirstTime = cglobal((:FirstTime, liblrsnash), Clong)
```

Comment:
@oyamad Do you know what we should do in lrslib v7.1a for this ? With this line, it complains that `FirstTime` is not defined and without this line, I have a test failing (see the test logs, not sure if that's related to this line or if that's another change).

Reply:
Yes, I believe we still need it: there is no change in `v7.1a`: https://github.com/JuliaPolyhedra/lrslib/blob/ab1a5313a54a6d9aadcaa2f2e8523856f747f7e1/lrsnashlib.c#L55

Isn't this as pointed out by @bzinberg in https://github.com/JuliaPolyhedra/lrslib/issues/2#issuecomment-823505797?
The declaration of `FirstTime` has been removed in `lrsnashlib.h` in `v7.1a`: https://github.com/JuliaPolyhedra/lrslib/commit/607673ef5eb7dd62d498b9ed7b88b0c983b306e5#diff-8325ef3172d6193ffabfcb846d744f86c389a21a61c802c4c715f55462857204L70

See https://github.com/bzinberg/lrslib/commit/453d0b0ad27f7d135ec993fe976652eb1287d2f9

Reply:
Agreed.  In case helpful, my thoughts as a newcomer to this codebase are:

* Changing the public API from an `extern long` that the user unsafe-sets to a `reset()` method is a good start.  But probably more important, the re-implementation of `lrs_solve_nash` in Julia [here](https://github.com/JuliaPolyhedra/LRSLib.jl/blob/78601f891fca396db98d2c7d1852a0d8d1814467/src/nash.jl#L45) seems like an anti-pattern to me.  It indicates that `FirstTime` really isn't a public API, it's just needed to enable the implementation to be split across the C codebase and the Julia codebase.  This is doable if it's needed, but perhaps we can revisit whether there's a way to bring all the logic into C with a documented API?
* `FirstTime` appears to be global state for the Nash solver.  I wonder if that state could be factored into a struct so that the logic in lrsnashlib could become stateless?
* (I'm not sure why we're using a `long` for what apparently has the semantics of a boolean, but I assume there is a good reason for that.)

Reply:
I agree. If the API was sufficient, we shouldn't have to define the `struct`s in Julia. However, it's sometimes difficult to bring the API to that level to the upstream package.
We already contacted David Avis trying to merge the changes we need upstream but we probably complicated things asking to have the build process work on more platforms which is now resolved and simplified thanks to the cross-compilation with Yggdrasil. We could aim at getting
https://github.com/JuliaPolyhedra/lrslib/commit/3962f8359cef6d69f47fe0a1686165433d930636
https://github.com/JuliaPolyhedra/lrslib/commit/ab1a5313a54a6d9aadcaa2f2e8523856f747f7e1
and
https://github.com/JuliaPolyhedra/lrslib/pull/3
to the next lrslib release and we could get rid of the fork.